src/context.rs

```diff
@@ -321,6 +337,138 @@ impl<C: Blockchain> MatchstickInstanceContext<C> {
             }
         }
 
+        let derived_fields = SCHEMA
+            .definitions
+            .iter()
+            .find_map(|def| {
+                if let schema::Definition::TypeDefinition(schema::TypeDefinition::Object(o)) = def {
+                    if o.name == entity_type {
+                        Some(o)
+                    } else {
+                        None
+                    }
+                } else {
+                    None
+                }
+            })
+            .unwrap_or_else(|| {
+                panic!(
+                    "{}",
+                    Log::Critical(format!("Something went wrong! Could not find the entity of type '{}' defined in the GraphQL schema.", &entity_type))
+                );
+            })
+            .fields
+            .iter()
```

Comment:
This seems to be pretty duplicated with [`let required_fields = ...`](https://github.com/LimeChain/matchstick/blob/9c4f3308c83f115798dd48951f8e90f34599fadb/src/context.rs#L284).

Maybe a common variable can be defined for the iterator over all fields and later be used to iterate over the required/derived fields accordingly.

Reply:
I would also suggest extracting some of the [common] logic [where/if possible] in separate functions, but I guess that can be done later in the refactoring sprint.